compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2332,7 +2332,15 @@ impl<'a> Parser<'a> {
             self.error_on_if_block_attrs(lo, false, block.span, attrs);
             block
         };
-        let els = if self.eat_keyword(kw::Else) { Some(self.parse_expr_else()?) } else { None };
+        let els = if self.eat_keyword(kw::Else) {
+            Some(self.parse_expr_else(false)?)
+        } else if self.may_recover() && self.token.is_keyword(sym::elif) {
```

Comment:
Doesn't this break code like this?

```rust
fn main() {
    let elif = ();
    if true {} elif
}
```

Reply:
Hmm, good point. I'll see if I can come up with a better solution